03_myfavorite/src/Components/GamePage.jsx

```diff
@@ -0,0 +1,118 @@
+import { React, useEffect, useState } from 'react';
+import Card from './Card';
+import ResultPage from './ResultPage';
+import 네오 from '../img/네오.jpg';
+import 라이언 from '../img/라이언.jpg';
+import 무지 from '../img/무지.jpg';
+import 어피치 from '../img/어피치.jpg';
+import 제이지 from '../img/제이지.jpg';
+import 춘식이 from '../img/춘식이.jpg';
+import 튜브 from '../img/튜브.jpg';
+import 프로도 from '../img/프로도.jpg';
+import vs from '../img/vs.png';
+import styled from 'styled-components';
+
+const gameInfo = [
+  { name: '네오', src: 네오 },
+  { name: '라이언', src: 라이언 },
+  { name: '무지', src: 무지 },
+  { name: '어피치', src: 어피치 },
+  { name: '제이지', src: 제이지 },
+  { name: '춘식이', src: 춘식이 },
+  { name: '튜브', src: 튜브 },
+  { name: '프로도', src: 프로도 },
+];
+
+function GamePage() {
+  const [leftCharacters, setLeftCharacters] = useState(gameInfo);
+  const [winnersInfo, setWinnersInfo] = useState([]);
+  const [currentRound, setCurrentRound] = useState(1);
+  const [rounds, setRounds] = useState(4);
+  const [winner, setWinner] = useState({});
+
+  function handleCardClick(friend) {
+    if (rounds === 1) {
+      setWinner(friend);
+    }
+    setLeftCharacters(leftCharacters.slice(2));
+    setWinnersInfo((prev) => [...prev, friend]);
+    setCurrentRound((prev) => prev + 1);
+  }
+
+  useEffect(() => {
+    if (leftCharacters.length === 0) {
+      setLeftCharacters(winnersInfo);
+      setWinnersInfo([]);
+      setCurrentRound(1);
+      setRounds((prev) => prev / 2);
+    }
+  }, [leftCharacters]);
+
+  if (rounds === 0.5) {
+    return <ResultPage winner={winner} />;
+  }
+
+  return (
+    <GameWrapper>
+      <WorldcupName>최고 귀여운 카카오프렌즈 월드컵</WorldcupName>
+      <RoundName>
+        Round: {currentRound}/{rounds}
+      </RoundName>
+      <CardWrapper>
+        {leftCharacters.map((friend, index) => {
+          if (index < 2) {
+            return (
+              <Card
+                key={friend.name}
+                name={friend.name}
+                src={friend.src}
+                onClick={() => {
+                  handleCardClick(friend);
+                }}
+              />
+            );
+          }
+          return null;
+        })}
```

Comment:
```suggestion
        {leftCharacters.slice(0, 1).map((friend, index) => (
              <Card
                key={friend.name}
                name={friend.name}
                src={friend.src}
                onClick={() => {
                  handleCardClick(friend);
                }}
              />
            );
          }
```

요렇게 써서 불필요한 분기처리를 줄일 수 있겠다!

Reply:
와 천재다